### PR TITLE
[11.0][mig] stock_valued_picking_report > stock_picking_report_valued

### PR DESCRIPTION
**Pull request by luismontalba**
Migration from V10.0 to V11.0
Module renamed to stock_picking_report_valued

Stock Picking Report Valued
=======================
Add amount information to Delivery Slip report. You can select at partner level if picking list report must be valued or not.

Configuration
-----------------

- Go to *Customers > (select one of your choice) > Sales & Purchases*.
- Set *Valued picking* field on.

Usage
--------

To get the stock picking valued report:

- Create a Sale Order with stockable products a *Valued picking* able
   customer.
- Confirm the Sale Order.
- Click on *Deliveries* stat button.
- Go to *Print > Delivery Slip*.

cc @Tecnativa

### CODE REVIEW

File: stock_picking_report_valued/__manifest__.py [hunk 1]
```diff
@@ -1,28 +1,26 @@
...
 # Copyright 2014 Pedro M. Baeza - Tecnativa <pedro.baeza@tecnativa.com>
 # Copyright 2015 Antonio Espinosa - Tecnativa <antonio.espinosa@tecnativa.com>
 # Copyright 2016 Carlos Dauden - Tecnativa <carlos.dauden@tecnativa.com>
 # Copyright 2017 David Vidal - Tecnativa <david.vidal@tecnativa.com>
+# Copyright 2017 Luis M. Ontalba - Tecnativa <luis.martinez@tecnativa.com>
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
 
 {
-    "name": "Stock Valued Picking Report",
+    "name": "Stock Picking Report Valued",
```
Comment (pedrobaeza) [inline stock_picking_report_valued/__manifest__.py]: It was correct before. We can remove `Stock` part if you want.

File: stock_picking_report_valued/__manifest__.py [hunk 2]
```diff
@@ -1,28 +1,26 @@
...
     "summary": "Adding Valued Picking on Delivery Slip report",
-    "version": "10.0.1.0.1",
+    "version": "11.0.1.0.1",
```
Comment (pedrobaeza) [inline stock_picking_report_valued/__manifest__.py]: 11.0.1.0.0

File: stock_picking_report_valued/models/stock_move_line.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
+# Copyright 2014 Pedro M. Baeza - Tecnativa <pedro.baeza@tecnativa.com>
+# Copyright 2015 Antonio Espinosa - Tecnativa <antonio.espinosa@tecnativa.com>
+# Copyright 2016 Carlos Dauden - Tecnativa <carlos.dauden@tecnativa.com>
+# Copyright 2018 Luis M. Ontalba - Tecnativa <luis.martinez@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class StockPackOperation(models.Model):
```
Comment (pedrobaeza) [inline stock_picking_report_valued/models/stock_move_line.py]: Rename to StockMoveLine

File: stock_picking_report_valued/models/stock_move_line.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        related='sale_line.price_subtotal', readonly=True,
+        string="Price subtotal")
+    sale_price_tax = fields.Float(
+        related='sale_line.price_tax', readonly=True,
+        string='Taxes')
+
+    @api.multi
+    @api.depends('sale_line.tax_id')
+    def _compute_sale_tax_description(self):
+        for operation in self:
```
Comment (pedrobaeza) [inline stock_picking_report_valued/models/stock_move_line.py]: s/operation/line

File: stock_picking_report_valued/models/stock_move_line.py [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+# Copyright 2018 Luis M. Ontalba - Tecnativa <luis.martinez@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class StockPackOperation(models.Model):
+    _inherit = "stock.move.line"
+
+    currency_id = fields.Many2one(
```
Comment (pedrobaeza) [inline stock_picking_report_valued/models/stock_move_line.py]: I think we don't need all this overload in the model. As they're related fields and they are not shown in any view, you can use dot notation for accessing its values in the report.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: stock_picking_report_valued/models/stock_move_line.py (hunk 5 above)
Comment (luismontalba) [reply stock_picking_report_valued/models/stock_move_line.py]: I've tried to remove it but test is telling me it is necessary. I think the monetary field needs this one as an auxiliary field

Comment (pedrobaeza) [reply stock_picking_report_valued/models/stock_move_line.py]: Well, I'm talking about all the fields, but OK, let's continue this way.

**Comment by luismontalba** [comment]
Changes done

**Review by pedrobaeza — APPROVED** [review]

**Comment by rafaelbn** [comment]
Testing in runbot I find something that maybe should be treated.

In the stock.picking , with all unit reserved but still not validated if you print the Delivery slip you get a PDF valued but incorrect because quantity is 0. Check screenshot,  you see valued with $ but how many? unit(s)



:thinking: 

Check this one http://3322527-44-613e08.runbot1.odoo-community.org/web#id=19&view_type=form&model=stock.picking&action=249&active_id=22

**Review by rafaelbn — COMMENTED** [review]
Tested with one doubt commented previously

**Comment by luismontalba** [comment]
I'll review it

File: stock_picking_report_valued/README.rst [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+   :alt: License: AGPL-3
+
+===========================
+Stock Picking Report Valued
+===========================
+
+Add amount information to Delivery Slip report.
+You can select at partner level if picking list report must be valued or not.
+If the picking is done it's valued with quantity done, otherwise the picking
+ is valued with reserved quantity.
```
Comment (pedrobaeza) [inline stock_picking_report_valued/README.rst]: Don't add this space at the beginning

Comment (luismontalba) [reply stock_picking_report_valued/README.rst]: Fixed (commit squashed)

**Comment by luismontalba** [comment]
@rafaelbn , now the picking is valued with done quantity when the state is done, and otherwise it's valued with reserved quantity.

File: stock_picking_report_valued/models/stock_move_line.py [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+            line.sale_tax_description = ', '.join([(
+                x.description or x.name) for x in line.sale_line.tax_id])
+
+    @api.depends('product_uom_qty', 'sale_discount', 'sale_price_unit',
+                 'sale_line.tax_id')
+    def _compute_amount(self):
+        for line in self:
+            price = line.sale_price_unit * (1 - (
+                line.sale_discount or 0.0) / 100.0)
+            if line.picking_id.state == 'done':
```
Comment (carlosdauden) [inline stock_picking_report_valued/models/stock_move_line.py]: Is needed this condition?
Don't change product_uom_qty  to qty_done when transfer is done?

**Review by carlosdauden — CHANGES_REQUESTED** [review]

File: stock_picking_report_valued/models/stock_move_line.py (hunk 7 above)
Comment (luismontalba) [reply stock_picking_report_valued/models/stock_move_line.py]: Well, depending on state of the picking, the quantity to take is in the field product_uom_qty or is in the field qty_done. This is the way I found to value the picking in both states, but any better way is welcome!

**Comment by pedrobaeza** [comment]
@carlosdauden what is missing here?

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**